datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -586,8 +587,47 @@ fn build_predicate_expression(
                 .min_column_expr()?
                 .lt_eq(expr_builder.scalar_expr().clone())
         }
+        Operator::Like => {
+            match &**right {
+                // If the literal is a 'starts_with'
+                Expr::Literal(ScalarValue::Utf8(Some(string)))
+                    if !string.starts_with('%') =>
+                {
+                    // Split the string to get the first part before '%'
+                    let split = string.split('%').next().unwrap().to_string();
```

Comment:
won't this `unwrap` panic if the string does not contain any `%`? (if "like" always requires that, maybe we should throw an error instead?)

Reply:
Like does not require `%`

Reply:
I initially wondered the same thing -- lol! But my conclusion was "no it won't panic"

I made a quick  playground that shows this working https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=049f4c1640386ff99c3b5e07085e0889

Reply:
Yes, won't panic because String::split() will always return at least 1 result, the full string if there's nothing to spilt by